Library/Homebrew/livecheck/strategy/cpan.rb

```diff
@@ -0,0 +1,67 @@
+# typed: false
+# frozen_string_literal: true
+
+require "uri"
+
+module Homebrew
+  module Livecheck
+    module Strategy
+      # The {Cpan} strategy identifies versions of software at
+      # cpan.metacpan.org by checking directory listing pages.
+      #
+      # CPAN URLs take the following format:
+      #
+      # * `https://cpan.metacpan.org/authors/id/M/MI/MIYAGAWA/Carton-v1.0.34.tar.gz`
+      #
+      # @api public
+      class Cpan
+        NICE_NAME = "CPAN"
+
+        # The allowlist used to determine if the strategy applies to the URL.
+        HOST_ALLOWLIST = ["cpan.metacpan.org"].freeze
+
+        # Whether the strategy can be applied to the provided URL.
+        #
+        # @param url [String] the URL to match against
+        # @return [Boolean]
+        def self.match?(url)
+          uri = URI.parse url
+          HOST_ALLOWLIST.include? uri.host
+        rescue URI::InvalidURIError
+          false
+        end
+
+        # Generates a URL and regex (if one isn't provided) and passes them
+        # to {PageMatch.find_versions} to identify versions in the content.
+        #
+        # @param url [String] the URL of the content to check
+        # @param regex [Regexp] a regex used for matching versions in content
+        # @return [Hash]
+        def self.find_versions(url, regex = nil)
+          %r{
+            /authors/id/(?<author_path>(?:[^/](?:/[^/]+){2})) # The author path (e.g. M/MI/MIYAGAWA)
```

Comment:
`(?:[^/](?:/[^/]+){2}))` looks a bit more "permissive" than necessary. Looking at https://cpan.metacpan.org/authors/id/, it looks like the second directory always has two characters, doesn't it. It's up to @samford to decide how stringent they want to be with this, but I'd use something like:
```
(?:[^/]/[^/]{2}/[^/]+))
```
but please don't rush changing what you have here until you hear from @samford.

Reply:
My suggestion below address this but I'll comment on this specifically to explain a bit.

Since some archive files are found in a subdirectory within the author directory (e.g., `/E/EX/EXAMPLE/example/example-v1.2.3.tar.gz`) I don't see any benefit to separating the author path (e.g., `/E/EX/EXAMPLE`) from the rest of the path (`/example`) before the filename. We can simply generate the page to check by using the full path before the filename (e.g., `/E/EX/EXAMPLE/example/`) without needing to separate it into parts.

Reply:
The reason for separation was to make it easier for the reader to grasp 🤷